src/Development/Shake/Internal/Rules/File.hs

```diff
@@ -58,6 +58,12 @@ newtype FileQ = FileQ {fromFileQ :: FileName}
 data FileA = FileA {-# UNPACK #-} !ModTime {-# UNPACK #-} !FileSize FileHash
     deriving (Typeable,Eq)
 
+-- | Result of a File rule, may contain raw file information and wether the rule did run this build
+data FileR = FileR { result :: Maybe FileA -- ^ Raw information about the file built by this rule. Set to 'Nothing' to prevent linting some times.
+                   , hasChanged :: Bool    -- ^ Wether the file changed this build. Transient information, that doesn't get serialized, is recreated at 'False' and doesn't participate in 'Eq', 'Hash' and the like.
```

Comment:
Spelling mistake - `Whether`. Probably also worth wrapping these lines at some point (I'm not a rigid follower of any particular length - but these lines are pretty long)